__tests__/app.test.js

```diff
@@ -1,30 +1,74 @@
-const db = require('../db/connection');
-const seed = require('../db/seeds/seed')
-const app = require('../app')
-const request = require("supertest")
+const db = require("../db/connection");
+const seed = require("../db/seeds/seed");
+const app = require("../app");
+const request = require("supertest");
 
 const index = require("../db/data/test-data/index");
 
 beforeEach(() => {
- return seed(index);
+  return seed(index);
 });
 
 afterAll(() => {
-db.end();
+  db.end();
 });
-
-describe.only("get /api/topics", () => {
+       //  -----    3    ------
+describe("Task-3 GET:/api/topics", () => {      
   test("responds with status 200", () => {
     return request(app).get("/api/topics").expect(200);
   });
+  test("Length of my topic should be greather then 0", () => {
+    return request(app)
+      .get("/api/topics")
+      .then((res) => {
+        const topics = res.body.topics;
+        expect(topics).not.toHaveLength(0);
+        // or
+        // expect(topic.length > 0).toBe(true);
+      });
+  });
   test("returns correct body", () => {
     return request(app)
       .get("/api/topics")
       .then((res) => {
-        console.log(">>>>TEST",res.body[0]);
-        const topic = res.body[0];
-        expect(topic).toHaveProperty("description");
-        expect(topic).toHaveProperty("slug");
+        const topics = res.body.topics;
+        topics.forEach(topic => {
+          expect(topic).toHaveProperty("description",expect.any(String));
+          expect(topic).toHaveProperty("slug",expect.any(String));
+        });
+      });
+  });
+});
+
+      //  -----    4    ------
+describe("Task-4 GET:/api/articles", () => {
+  test("respons with status 200", () => {
+    return request(app).get("/api/articles").expect(200);
+  });
+  test(`An article should have the follow property:
+          - propertyaurhot,
+          - title,
+          - article_id,
+          - topic_createe_at,
+          - votes,
+          - article_img_url,
+          - comment_count`
+    , () => {
+    return request(app)
+      .get("/api/articles")
+      .then((res) => {
+        const articles = res.body.articles;
+        expect(articles.length > 0).toBe(true);
+        articles.forEach((article) => {
+          expect(article).toHaveProperty("author");
```

Comment:
We should be checking the type of the value for these properties. Like we did in the tests for the topics.